src/sources/mod.rs

```diff
@@ -106,144 +106,180 @@ enum BuildError {
+    #[configurable(metadata(docs::label = "Demo Logs"))]
     DemoLogs(demo_logs::DemoLogsConfig),
 
-    /// DNSTAP.
+    /// Collect DNS logs from a dnstap-compatible server.
     #[cfg(all(unix, feature = "sources-dnstap"))]
+    #[configurable(metadata(docs::label = "dnstap"))]
     Dnstap(dnstap::DnstapConfig),
 
-    /// Docker Logs.
+    /// Collect logs from Docker.
     #[cfg(feature = "sources-docker_logs")]
+    #[configurable(metadata(docs::label = "Docker Logs"))]
     DockerLogs(docker_logs::DockerLogsConfig),
 
-    /// EventStoreDB Metrics.
+    /// Receive metrics from collected by a EventStoreDB.
     #[cfg(feature = "sources-eventstoredb_metrics")]
+    #[configurable(metadata(docs::label = "EventStoreDB Metrics"))]
     EventstoredbMetrics(eventstoredb_metrics::EventStoreDbConfig),
 
-    /// Exec.
+    /// Collect output from a process running on the host.
     #[cfg(feature = "sources-exec")]
+    #[configurable(metadata(docs::label = "Exec"))]
     Exec(exec::ExecConfig),
 
-    /// File.
+    /// Collect logs from files.
     #[cfg(feature = "sources-file")]
+    #[configurable(metadata(docs::label = "File"))]
     File(file::FileConfig),
 
-    /// File descriptor.
+    /// Collect logs from a file descriptor.
     #[cfg(all(unix, feature = "sources-file-descriptor"))]
+    #[configurable(metadata(docs::label = "File Descriptor"))]
     FileDescriptor(file_descriptors::file_descriptor::FileDescriptorSourceConfig),
 
-    /// Fluent.
+    /// Collect logs from a Fluentd or Fluent Bit agent.
     #[cfg(feature = "sources-fluent")]
+    #[configurable(metadata(docs::label = "Fluent"))]
     Fluent(fluent::FluentConfig),
 
-    /// GCP Pub/Sub.
+    /// Fetch observability events from GCP's Pub/Sub messaging system.
     #[cfg(feature = "sources-gcp_pubsub")]
+    #[configurable(metadata(docs::label = "GCP Pub/Sub"))]
     GcpPubsub(gcp_pubsub::PubsubConfig),
 
-    /// Heroku Logs.
+    /// Collect logs from Heroku's Logplex, the router responsible for receiving logs from your Heroku apps.
     #[cfg(feature = "sources-heroku_logs")]
+    #[configurable(metadata(docs::label = "Heroku Logplex"))]
     HerokuLogs(heroku_logs::LogplexConfig),
 
-    /// Host Metrics.
+    /// Collect metric data from the local system.
     #[cfg(feature = "sources-host_metrics")]
+    #[configurable(metadata(docs::label = "Host metrics"))]
     HostMetrics(host_metrics::HostMetricsConfig),
 
-    /// HTTP.
+    /// Receive logs emitted by an HTTP server.
     #[cfg(feature = "sources-http_server")]
+    #[configurable(deprecated)]
+    #[configurable(metadata(docs::label = "HTTP"))]
     Http(http_server::HttpConfig),
 
-    /// HTTP Client.
+    /// Pull observability data from an HTTP server at a configured interval.
     #[cfg(feature = "sources-http_client")]
+    #[configurable(metadata(docs::label = "HTTP Client"))]
     HttpClient(http_client::HttpClientConfig),
 
-    /// HTTP Server.
+    /// Receive logs emitted by an HTTP server.
```

Comment:
```suggestion
    /// Receive logs emitted by an HTTP client.
```

**suggestion**: Given Vector is acting as the "server" in the client/server paradigm here, I'd change the wording.